miniml/compiler/compile.scm

```diff
@@ -601,10 +613,18 @@
     )))
 
 (define (token errorp)
-  (let* ((location (make-source-location "*stdin*" (port-line (current-input-port)) (port-column (current-input-port)) -1 -1))
+  (let* ((location (current-location))
          (c (read-char)))
     (token-dispatch errorp location c)))
 
+(define (try-read offset input)
```

Comment:
I think this function is no longer needed.

Reply:
It's a pity, it was fun to write.